ingest/indexer/service/indexer_streaming_service_test.go

```diff
@@ -0,0 +1,423 @@
+			expectedLiquidity:   true,
+		},
+		{
+			name:                "non token_swapped event",
+			eventType:           gammtypes.TypeEvtPoolJoined,
+			havePoolIDAttribute: true,
+			useCLPoolID:         true,
+			expectedError:       false,
+			expectedLiquidity:   false,
+		},
+		{
+			name:                "use non existent pool_id",
+			eventType:           gammtypes.TypeEvtTokenSwapped,
+			havePoolIDAttribute: true,
+			useCLPoolID:         false,
+			expectedError:       true,
+			expectedLiquidity:   false,
+		},
+		{
+			name:                "no pool_id attribute in event",
+			eventType:           gammtypes.TypeEvtTokenSwapped,
+			havePoolIDAttribute: false,
+			useCLPoolID:         true,
+			expectedError:       true,
+			expectedLiquidity:   false,
+		},
+	}
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.Setup()
+
+			// This test suite is to test the AddTokenLiquidity method in the indexer streaming service.
+			// where it applies to: token_swapped event only, i.e. gammtypes.TypeEvtTokenSwapped
+			// It then looks for the pool_id attribute (concentratedliquiditytypes.AttributeKeyPoolId) in the
+			// event attribute map.  With the pool_id, it then fetches the pool liquidity thru
+			// keepers.PoolManagerKeeper.GetTotalPoolLiquidity function. The pool liquidity data is then appended
+			// to the event attribute map with the key "liquidity_{denom}", value being the pool liquidity.
+
+			// Initialized chain pools
+			s.PrepareAllSupportedPools()
+
+			// Get all chain pools from state for asserting later
+			concentratedPools, err := s.App.ConcentratedLiquidityKeeper.GetPools(s.Ctx)
+
+			s.Require().NoError(err)
+
+			cfmmPools, err := s.App.GAMMKeeper.GetPools(s.Ctx)
+			s.Require().NoError(err)
+
+			cosmWasmPools, err := s.App.CosmwasmPoolKeeper.GetPoolsWithWasmKeeper(s.Ctx)
+			s.Require().NoError(err)
+
+			blockPools := commondomain.BlockPools{
+				ConcentratedPools: concentratedPools,
+				CFMMPools:         cfmmPools,
+				CosmWasmPools:     cosmWasmPools,
+			}
+
+			transformedPools := []sqsdomain.PoolI{}
+			for _, pool := range blockPools.GetAll() {
+				// Note: balances are irrelevant for the test so we supply empty balances
+				transformedPool := sqsdomain.NewPool(pool, pool.GetSpreadFactor(s.Ctx), sdk.Coins{})
+				transformedPools = append(transformedPools, transformedPool)
+			}
+
+			// Initialize an empty pool tracker
+			emptyPoolTracker := pooltracker.NewMemory()
+
+			// Initialize a mock pool extractor
+			poolExtractorMock := &sqsmocks.PoolsExtractorMock{
+				BlockPools: commondomain.BlockPools{
+					ConcentratedPools: concentratedPools,
+					CFMMPools:         cfmmPools,
+					CosmWasmPools:     cosmWasmPools,
+				},
+			}
+
+			// Initialize a block process strategy manager
+			blockProcessStrategyManager := commondomain.NewBlockProcessStrategyManager()
+
+			// Initialize a concentrated pool with spread factor
+			sf, _ := math.LegacyNewDecFromStr(defaultSpreadFactor)
```

Comment:
nit: let's use `osmomath.NewDecFromStr` for consistency with the rest of the codebase

Reply:
Done 👍